lib/app/dispatcher.rb

```diff
@@ -55,12 +59,21 @@ def pool
         )
       end
 
-      def scheduler
-        @scheduler ||= Core::NativeScheduler.new(POLL_IDLING)
+      def sync_scheduler
+        @sync_scheduler ||= Core::NativeScheduler.new(:sync, POLL_IDLING)
+      end
+
+      def heartbeat_scheduler
+        @heartbeat_scheduler ||= Core::NativeScheduler.new(:heartbeat, POLL_IDLING)
+      end
+
+      def configuration_scheduler
+        @configuration_scheduler ||= Core::NativeScheduler.new(:configuration, 5)
```

Comment:
Why do we reuse the `POLL_IDLING` for the heartbeat and sync, but hardcode the config scheduler? If the idle intervals should be different, wouldn't it still make sense to make the other one configurable, too?

Reply:
Yes, again, will make them configurable once I get feedback from the team.